### ISSUE TITLE
Behavior of Shift+F7 and Alt+F7 in Viewer and Editor

### ISSUE THREAD

**Issue by MKadaner** [FarGroup/FarManager#600]
### Description of the new feature or improvement

Currently, the search direction of Shift+F7 and Alt+F7 depends on the "Reverse search" option of the Search dialog, viz.
- Shift+F7 continues searching in the _direction set in the Search dialog._
- Alt+F7 continues searching in the _direction opposite to that set in the Search dialog._

Somehow, this behavior confuses me to no end. In fact, because of this behavior, I never check "Reverse search" checkbox in the dialog, and use Shift+F7 and Alt+F7 hotkeys as if they search in a certain direction, no matter the dialog option:
- Shift+F7 continues searching forward.
- Alt+F7 continues searching backward.

This effectively means that if I want to search backward, towards the beginning of the file, I open the dialog, leave "Reverse search" checkbox unchecked, press "Enter" and as soon as the next pattern is found, press Alt+F7 to find the previous pattern. I then proceed pressing Alt+F7 if I want to continue searching.

I would like to have a global "far:config" parameter controlling this behavior. By default (`false`), the current behavior is used. If the parameter is set to `true`, the new behavior is activated. The parameter could be called something like "SearchContinueAbsoluteDirection".

### Proposed technical implementation details (optional)

Though Editor / Viewer search implementation is a bit tricky, I think I can do this feature. Any objections?

**Comment by alabuzhev** [comment]
> If the parameter is set to `true`, the new behavior is activated

So what exactly the new behavior should be?
Shift+F7 always forward and Alt+F7 always backward, `[x] Reverse search` affects only `F7 Enter`?

**Comment by MKadaner** [comment]
That is correct. `[x] Reverse search` becomes somewhat less useful. I would use it in a scenario where I am in the middle of a large log (think millions of lines) looking for something relatively rare, just a few occurrences. I know that the place of interest is above the current position--I want to find the instance of this something which happened before here, where I am observing the consequences. If I `F7`, `[ ] Reverse search`, `Enter`, as I usually do, I first jump forward to an undefined and irrelevant position and risk losing the context of my operation. Sometimes, I literally close my eyes before pressing `Enter` and immediately press `Alt+F7`.

The current behavior is especially vexing in a situation when I was tracking something for some time jumping back and forth multiple times with `Shift+F7` / `Alt+F7`, then did something else, maybe even switched to another task, then want to find the next occurrence in a certain direction. I know that the search string is still what I am looking for, but I absolutely cannot remember what the value of `Reverse search` was. I should either open the dialog again only to set the right search direction, or continue searching with 50% chance of getting to a wrong place.

**Comment by alabuzhev** [comment]
It is confusing indeed. I also never use the checkbox and either press enter and then Alt+F7 or jump to the beginning before searching.

I'm not a big fan of new far:config options though, maybe we can do it without them?

A couple of alternative solutions:
- Automatically uncheck the `[x] Reverse search` when the dialog is closed.
`F7 Enter` searches backward, afterwards Shift+F7 searches forward, Alt+F7 - backward.
- Remove the checkbox and replace the `{ Search }` button with `{ Search down } [ Search up ]`.
The initial direction depends on the pressed button, afterwards Shift+F7 searches forward, Alt+F7 - backward.

**Comment by MKadaner** [comment]
Of the two solutions, I certainly prefer the latter, with two buttons. Auto-clearing `[x] Reverse search` will be also confusing. Most other dialog options in many dialogs are preserved between dialog invocations.

Let me finish Text / Hex controls on the standard search dialog, then I will implement `{ Search down } [ Search up ]` buttons on it (if there are no better ideas by that time).

By the way, how about something like this:
```
{ ⇓ Search ⇓ } [ ⇑ Search ⇑ ]
```
with the hotkeys `Alt+Down` and `Alt+Up` respectfully?

**Comment by alabuzhev** [comment]
I'd probably prefer the latter too as more obvious.

> `{ ⇓ Search ⇓ } [ ⇑ Search ⇑ ]`

This probably will be unreadable with raster fonts (yes, there are people who still use them for some reason).

**Comment by MKadaner** [comment]
> there are people who still use [raster fonts]

Ugh!

The words it is! Less hassle implementing non-standard behavior.

**Comment by HamRusTal** [comment]
> Remove the checkbox and replace the `{ Search }` button with `{ Search down } [ Search up ]`.

I'm definitely for this solution. And plain words are better than fancy symbols as words are easier to communicate  to another person.

**Comment by rohitab** [comment]
> Remove the checkbox and replace the `{ Search }` button with `{ Search down } [ Search up ]`.
> The initial direction depends on the pressed button, afterwards Shift+F7 searches forward, Alt+F7 - backward.

This is definitely more user-friendly, and similar to how it's implemented in other editors, like Visual Studio.

**Comment by MKadaner** [comment]
@alabuzhev, assign this one to me please.
